torchvision/transforms.py

```diff
@@ -365,15 +414,16 @@ def __call__(self, img):
                 w, h = h, w
 
             if w <= img.size[0] and h <= img.size[1]:
-                x1 = random.randint(0, img.size[0] - w)
-                y1 = random.randint(0, img.size[1] - h)
-
-                img = img.crop((x1, y1, x1 + w, y1 + h))
-                assert(img.size == (w, h))
-
-                return img.resize((self.size, self.size), self.interpolation)
+                x = random.randint(0, img.size[0] - w)
+                y = random.randint(0, img.size[1] - h)
+                return x, y, w, h
 
         # Fallback
-        scale = Scale(self.size, interpolation=self.interpolation)
-        crop = CenterCrop(self.size)
-        return crop(scale(img))
+        w = min(img.size[0], img.shape[1])
+        x = (img.shape[0] - w) // 2
+        y = (img.shape[1] - w) // 2
+        return x, y, w, w
+
+    def __call__(self, img):
+        x, y, w, h = self.get_params(img)
+        return scaled_crop(img, x, y, w, h, self.size, self.interpolation)
```

Comment:
Instead of calling `scaled_crop`, can't we instead call `crop`+`scale`?

Reply:
I agree, I think it would be cleaner and not clutter the interface as much

Reply:
This is why I did this: (replied before)
> I've considered this. But added scaled_crop because I want user to be able to create joint randomsized crop with as little hassle as possible. Otherwise user need to know exactly how it works.

Do you still think we should use crop + scale instead of scaled_crop?